Geometry/HGCalCommonData/plugins/DDHGCalEEFileAlgo.cc

```diff
@@ -0,0 +1,425 @@
+                                 rMaxFront_,
+                                 slopeT_,
+                                 -layerSense_[ly],
+                                 pgonZ,
+                                 pgonRin,
+                                 pgonRout);
+#ifdef EDM_ML_DEBUG
+          edm::LogVerbatim("HGCalGeom") << "DDHGCalEEFileAlgo: z " << (zz - hthick) << ":" << (zz + hthick) << " with "
+                                        << pgonZ.size() << " palnes";
+          for (unsigned int isec = 0; isec < pgonZ.size(); ++isec)
+            edm::LogVerbatim("HGCalGeom")
+                << "[" << isec << "] z " << pgonZ[isec] << " R " << pgonRin[isec] << ":" << pgonRout[isec];
+#endif
+          for (unsigned int isec = 0; isec < pgonZ.size(); ++isec) {
+            pgonZ[isec] -= zz;
+            pgonRout[isec] = pgonRout[isec] * cosAlpha_ - tol;
+          }
+        }
+        DDSolid solid =
+            DDSolidFactory::polyhedra(DDName(name, nameSpace_), sectors_, -alpha_, 2._pi, pgonZ, pgonRin, pgonRout);
+        glog = DDLogicalPart(solid.ddname(), matter, solid);
+#ifdef EDM_ML_DEBUG
+        edm::LogVerbatim("HGCalGeom") << "DDHGCalEEFileAlgo: " << solid.name() << " polyhedra of " << sectors_
+                                      << " sectors covering " << convertRadToDeg(-alpha_) << ":"
+                                      << convertRadToDeg(-alpha_ + 2._pi) << " with " << pgonZ.size()
+                                      << " sections and filled with " << matName << ":" << &matter;
+        for (unsigned int k = 0; k < pgonZ.size(); ++k)
+          edm::LogVerbatim("HGCalGeom") << "[" << k << "] z " << pgonZ[k] << " R " << pgonRin[k] << ":" << pgonRout[k];
+#endif
+      } else {
+        DDSolid solid = DDSolidFactory::tubs(DDName(name, nameSpace_), hthick, rinB, routF, 0.0, 2._pi);
+        glog = DDLogicalPart(solid.ddname(), matter, solid);
+#ifdef EDM_ML_DEBUG
+        edm::LogVerbatim("HGCalGeom") << "DDHGCalEEFileAlgo: " << solid.name() << " Tubs made of " << matName << ":"
+                                      << &matter << " of dimensions " << rinB << ", " << routF << ", " << hthick
+                                      << ", 0.0, 360.0 and position " << glog.name() << " number " << copy << ":"
+                                      << layerCenter_[copy - 1];
+#endif
+        positionSensitive(glog, rinB, routF, zz, layerSense_[ly], (copy - 1), cpv);
+      }
+      DDTranslation r1(0, 0, zz);
+      DDRotation rot;
+      cpv.position(glog, module, copy, r1, rot);
+      ++copyNumber_[ii];
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HGCalGeom") << "DDHGCalEEFileAlgo: " << glog.name() << " number " << copy << " positioned in "
+                                    << module.name() << " at " << r1 << " with " << rot;
+#endif
+      zz += hthick;
+    }  // End of loop over layers in a block
+    zi = zo;
+    laymin = laymax;
+    if (std::abs(thickTot - layerThick_[i]) < 0.00001) {
```

Comment:
Need a comment here for an empty `if` block.

Reply:
Done